src/components/modals/MigrateV1Pt1Modal.tsx

```diff
@@ -0,0 +1,104 @@
+import { Button, Modal } from 'antd'
+import { ProjectContext } from 'contexts/projectContext'
+import { UserContext } from 'contexts/userContext'
+import { useContext, useState } from 'react'
+import { getTerminalAddress } from 'utils/terminal-versions'
+
+export default function MigrateV1Pt1Modal({
+  visible,
+  onCancel,
+}: {
+  visible: boolean
+  onCancel: VoidFunction
+}) {
+  const [loadingAddToBalance, setLoadingAddToBalance] = useState<boolean>()
+  const [loadingMigrate, setLoadingMigrate] = useState<boolean>()
+  const { contracts, transactor } = useContext(UserContext)
+  const { projectId, balance, handle } = useContext(ProjectContext)
+
+  const needsBalance = balance?.eq(0)
+
+  function migrate() {
+    const terminalV1_1Address = getTerminalAddress('1.1')
+
+    if (!transactor || !contracts || !projectId || !terminalV1_1Address) return
+
+    setLoadingMigrate(true)
+
+    transactor(
+      contracts.TerminalV1,
+      'migrate',
+      [projectId.toHexString(), terminalV1_1Address],
+      {
+        onDone: () => {
+          setLoadingMigrate(false)
+          onCancel()
+        },
+      },
+    )
+  }
+
+  function add1Gwei() {
+    if (!transactor || !contracts || !projectId) return
+
+    setLoadingAddToBalance(true)
+
+    transactor(
+      contracts.TerminalV1,
+      'addToBalance',
+      [projectId.toHexString()],
+      {
+        value: '0x01',
+        onDone: () => setLoadingAddToBalance(false),
+      },
+    )
+  }
+
+  return (
+    <Modal
+      visible={visible}
+      onOk={migrate}
+      onCancel={onCancel}
+      okText="Migrate to V1.1"
+      okType="primary"
+      confirmLoading={loadingMigrate}
+      okButtonProps={{ disabled: needsBalance }}
+    >
+      <h2>Migrate to v1.1</h2>
+      <p>
+        This project is currently using the Juicebox v1 terminal. New features
```

Comment:
**suggestion**: does the user typically know what `terminal` means? I wonder if it's okay to say something more general, like:

```suggestion
        This project is currently deployed on the Juicebox v1 smart contracts. New features
```

Reply:
ehh probably not. but it also may be misleading to say "v1 smart contracts", since it's really only the TerminalV1 contract being updated. i think i'd rather err on the side of clarity and encourage folks to ask "what's a terminal?"